src/Constants.py

```diff
@@ -1,22 +1,22 @@
 from enum import Enum
 
 EXIT_PAYMENT_TYPE = "exit"
-
-CURRENT_TESTNET = 'GRANADANET'
+PROTOCOL_NAME = 'GRANADA'
+CURRENT_TESTNET = '{}NET'.format(PROTOCOL_NAME)
 
 FIRST_GRANADA_LEVEL = 1589249
 
 # Providers api prefix
 # Public RPC
 PUBLIC_NODE_URL = {
     'MAINNET': 'https://mainnet-tezos.giganode.io',
-    CURRENT_TESTNET: 'https://granadanet.smartpy.io'
+    CURRENT_TESTNET: 'https://{}.smartpy.io'.format(CURRENT_TESTNET)
```

Comment:
This is the culprit. The url` https://granadanet.smartpy.io` returns 502 error. Means it has been taken down.